test/filter.test.ts

```diff
@@ -90,3 +90,8 @@ expectType<(number | undefined)[]>(filter(x => x != null, [] as (number | undefi
 // filter(() => narrow, dist)
 // no need for type annotations when using full signature
 expectType<Record<string, string>>(filter(isNotNil, {} as Dict));
+
+
+// filter(() => narrow)(dist)
+expectType<Map<string, string>>(filter(isNotNil, new Map<string, string | undefined>()));
+expectType<Map<string, number>>(filter(gt5, new Map<string, number>()));
```

Comment:
First, these tests are for `// filter(() => narrow, map)`

Second, you need to be far more exhaustive with your test additions. This file has test coverage for all curry varieties and argument type overloads. You added support for a new argument type, so you need to add tests for each curry call variation